File: leetcode/math/573_Squirrel_Simulation/codes/math.py
from typing import List

class Solution:
    def minDistance(self, height: int, width: int, tree: List[int], squirrel: List[int], nuts: List[List[int]]) -> int:
        total = 0
        max_saving = float("-inf")

        for nut in nuts:
            dist_tree = abs(nut[0] - tree[0]) + abs(nut[1] - tree[1])
            dist_squirrel = abs(nut[0] - squirrel[0]) + abs(nut[1] - squirrel[1])
            total += 2 * dist_tree
            max_saving = max(max_saving, dist_tree - dist_squirrel)

        return total - max_saving

File: leetcode/math/573_Squirrel_Simulation/codes/math.c
#include <stdlib.h>
#include <limits.h>

int absDiff(int a, int b) {
    return (a > b) ? (a - b) : (b - a);
}

int minDistance(int height, int width, int* tree, int treeSize, int* squirrel, int squirrelSize, int** nuts, int nutsSize, int* nutsColSize) {
    int total = 0;
    int maxSaving = INT_MIN;

    for (int i = 0; i < nutsSize; ++i) {
        int* nut = nuts[i];
        int distTree = absDiff(nut[0], tree[0]) + absDiff(nut[1], tree[1]);
        int distSquirrel = absDiff(nut[0], squirrel[0]) + absDiff(nut[1], squirrel[1]);
        total += 2 * distTree;
        int saving = distTree - distSquirrel;
        if (saving > maxSaving) {
            maxSaving = saving;
        }
    }

    return total - maxSaving;
}
